### PR TITLE
HPCC-32005 Documentation for the hyperlinks testing GitHub Action

### PR DESCRIPTION
**Pull request by Charan-Sharan**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: .github/workflows/test-hyperlinks.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  run: |
+    for LINE in $( cat internalLinks.txt )
+    do 
+      REFERENCE=$( echo $LINE | cut -d ':' -f3- )
+      FILE=$( echo $LINE | cut -d ':' -f1 )
+      if [[ ${REFERENCE:0:1} == '#' ]]; then 
+        Link_text=$( cat $FILE | grep  -oE "\[.*\]\(${REFERENCE}\)" | sed 's/\[//' | cut -d ']' -f1 )
+        IS_PRESENT=$(cat $FILE | grep -oE "# ${Link_text}" | wc -w)
+        if [[ $IS_PRESENT -eq 0 ]]; then 
+				  echo -e "${LINE} -\u001b[31m invalid reference"
```
Comment (JamesDeFabia) [inline .github/workflows/test-hyperlinks.md:192]: Not sure what the red bars are here and on line 204. They show on the rendered page as well.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Looks good from my POV except for the two places with red bars, noted below

File: .github/workflows/test-hyperlinks.md (hunk 1 above)
Comment (Charan-Sharan) [reply .github/workflows/test-hyperlinks.md]: Removed those red bars.

**Review by JamesDeFabia — APPROVED** [review]
Looks good from my POV. Nice work

**Review by AttilaVamos — APPROVED** [review]
Seems good to go.

**Comment by GordonSmith** [comment]
@AttilaVamos would `devdoc` be a better location for this document?

**Comment by AttilaVamos** [comment]
@GordonSmith let's ask @JamesDeFabia and @stuartort about "would devdoc be a better location for this document?"

**Comment by JamesDeFabia** [comment]
Since we have other MD files in the \workflow\ folder, I think this is a good place.

**Comment by GordonSmith** [comment]
The other ones are used by GitHub for things like the default PR comment etc.?

**Comment by JamesDeFabia** [comment]
There is one with instructions for Bundle Testing

**Review by JamesDeFabia — APPROVED** [review]
Looks good from my POV

**Review by AttilaVamos — APPROVED** [review]
Seems good.

**Pull request merged**